optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -364,25 +417,153 @@ def get_hub_cached_entries(
+            for param in white_list:
+                value.pop(param, None)
+                target_entry_config[name].pop(param, None)
+            for term in set(entry_config[name]).intersection(set(target_entry_config[name])):
+                if entry_config[name][term] != target_entry_config[name][term]:
+                    is_matched = False
+        else:
+            if value != target_entry_config[name]:
+                is_matched = False
+    if is_matched:
+        neuron_config.pop("model_type", None)
+        model_entries.append(neuron_config)
+
+    return model_entries
+
+
+def _prepare_config_for_matching(entry_config, target_entry, model_type):
+    if model_type == "stable-diffusion":
+        neuron_config = entry_config["unet"].pop("neuron")
+        non_checked_components = [
+            "vae",
+            "vae_encoder",
+            "vae_decoder",
+        ]  # Exclude vae configs from the check for now since it's complex and not mandatory
+        for param in non_checked_components:
+            entry_config.pop(param, None)
+            target_entry.config.pop(param, None)
+        target_entry_config = target_entry.config
+    else:
+        neuron_config = entry_config.pop("neuron")
+        entry_config = {"model": entry_config}
+        target_entry_config = {"model": target_entry.config}
+
+    return entry_config, target_entry_config, neuron_config
+
+
+def get_multimodels_configs_from_hub(model_id):
+    api = HfApi()
+    repo_files = api.list_repo_files(model_id)
+    config_pattern = "/config.json"
+    config_files = [path for path in repo_files if config_pattern in path]
+    lookup_configs = {}
+    with TemporaryDirectory() as tmpdir:
+        for model_path in config_files:
+            local_path = api.hf_hub_download(model_id, model_path, local_dir=tmpdir)
+            with open(local_path) as f:
+                entry_config = json.load(f)
+                white_list = CACHE_WHITE_LIST
                 for param in white_list:
                     entry_config.pop(param, None)
-                    target_entry.config.pop(param, None)
-                if entry_config == target_entry.config:
-                    model_entries.append(neuron_config)
-    return model_entries
+                lookup_configs[model_path.split("/")[-2]] = entry_config
+
+    if "unet" in lookup_configs:
+        lookup_configs["model_type"] = "stable-diffusion"
+    return lookup_configs
+
+
+def exclude_white_list_from_config(
+    config: Dict, white_list: Optional[List] = None, neuron_white_list: Optional[List] = None
+):
+    if white_list is None:
+        white_list = CACHE_WHITE_LIST
+
+    if neuron_white_list is None:
+        neuron_white_list = NEURON_CONFIG_WHITE_LIST
+
+    for param in white_list:
+        config.pop(param, None)
+
+    for param in neuron_white_list:
+        config["neuron"].pop(param, None)
+
+    return config
+
+
+# Only applied on traced TorchScript models
```

Comment:
I would more add a docstring than a comment like that.

Reply:
ok! will change it to docstring.